### ISSUE TITLE
`createTableLike` does not execute the callback

### ISSUE THREAD

**Issue by bakasmarius** [knex/knex#4820]
# Environment

Knex version: 0.95.12; 0.95.14

# Bug

`createTableLike` only copies the old table, but does not add the extra columns from the callback.

```
knex.schema.createTableLike('new_users', 'users', (table) => {
    table.text('new_column')
  })
```
results in the following query: `create table "new_users" (like "users" including all);`.
Expected behavior: it results in `create table "new_users" (like "users" including all, "new_column" text);`

**Comment by OlivierCavadenti** [comment]
It's a typing issue actually.
The current implementation of createTableLike dont allow callback and just copy the old table (documentation dont talk about callback function https://knexjs.org/#Schema-createTableLike).
By the way, it could be useful to improve createTableLike to take care about this callback (need special treatment because the callback dont modify the first table, but the duplicate column).

**Comment by bakasmarius** [comment]
Yes, the documentation code wasn't working for me, so from the typing, I assumed it works the same as `createTable`.
Would it be possible to get the functionality I was actually expecting?
It's quite useful for creating base tables that consist of common columns like `id`, `created_at`, `updated_at`, etc., and using them as a base for creating all the rest of the tables.
Without the ability to extend the copied table with extra columns, the functionality is not THAT useful/flexible IMHO.

**Comment by OlivierCavadenti** [comment]
>  Would it be possible to get the functionality I was actually expecting?
Yes, I can take a look next week.

**Comment by kibertoad** [comment]
Released in 1.0.1.
